File: server/Models/Emotion.cs
using System.ComponentModel.DataAnnotations;

namespace FeelJournal.Models;

public class Emotion
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(255)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(255)]
    public string Icon { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<EmotionEntry> EmotionEntries { get; set; } = new List<EmotionEntry>();
}

File: app/src/main/java/com/example/feel_journal/data/model/Emotion.kt
package com.example.feel_journal.data.model

import com.google.gson.annotations.SerializedName

data class Emotion(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("icon") val icon: String,
    @SerializedName("createdAt") val createdAt: String
)
